specutils/spectra/spectrum_collection.py

```diff
@@ -0,0 +1,180 @@
+import logging
+from collections import MutableSequence
+
+import astropy.units as u
+import numpy as np
+from astropy.nddata import NDIOMixin, NDUncertainty
+
+from .spectrum1d import Spectrum1D
+
+__all__ = ['SpectrumArray', 'SpectrumCollection', 'ResampleMixin']
+
+
+class SpectrumArray(MutableSequence):
+    """
+    An array class to hold a set of :class:`~specutils.Spectrum1D` objects.
+    This class does not implicitly attempt to resample onto a new wavelength
+    grid.
+    """
+    def __init__(self, items):
+        self._items = list(items)
+
+    def __setitem__(self, index, value):
+        self._items[index] = value
+
+    def __delitem__(self, index):
+        del self._items[index]
+
+    def insert(self, index, value):
+        self._items.insert(index, value)
+
+    def __getitem__(self, index):
+        return self._items[index]
+
+    def __len__(self):
+        return len(self._items)
+
+
+class ResampleMixin:
+    """
+    Implements resampling behavior as a mixin class.
+    """
+    def deposit(self, grid):
+        """
+        Deposit spectrum list onto a new dispersion grid.
+
+        Parameters
+        ----------
+        grid : str, array-like, tuple
+            Define the dispersion grid to which all spectra in the collection
+            will be resampled. If 'fine', the smallest dispersion bin in all
+            spectra will be user; if 'coarse', then the largest will be used.
+            Users can also supply a custom dispersion array, or a tuple of the
+            form (start_bin, end_bin, bin_shape).
+        """
+        if isinstance(grid, np.ndarray):
+            resample_grid = grid
+        elif isinstance(grid, list):
+            resample_grid = np.array(grid)
+        elif isinstance(grid, tuple):
+            resample_grid = np.arange(*grid)
+        else:
+            if isinstance(grid, str) and len(self._items) > 0:
+                start_bin = min([x.spectral_axis[0] for x in self._items]).value
+                end_bin = max([x.spectral_axis[-1] for x in self._items]).value
```

Comment:
Note that these are the 'centers' of the start/end bins.  You probably instead want something like:
```
start_bin = min([x.spectral_axis[0]-np.abs((x.spectral_axis[1]-x.spectral_axis[0])/2) for x in self._items]).value
```
to get the edges.  This is important since you're optionally changing the bin size below.